File: common/src/main/java/com/cainsgl/common/util/UserUtils.java
package com.cainsgl.common.util;

import cn.dev33.satoken.stp.StpUtil;
import com.cainsgl.common.entity.user.UserEntity;

public class UserUtils
{
    private static final String USER_INFO_KEY = "userInfo";

    public static UserEntity getUserInfo(){
        Object o = StpUtil.getSession().get(USER_INFO_KEY);
        if(o!=null)
        {
            return (UserEntity)o;
        }
        return null;
    }
    public static void setUserInfo(UserEntity userEntity){
        StpUtil.getSession().set(USER_INFO_KEY,userEntity);
    }
}

File: common/src/main/java/com/cainsgl/common/util/UserUtils.kt
package com.cainsgl.common.util

import cn.dev33.satoken.stp.StpUtil
import com.cainsgl.common.entity.user.UserEntity

object UserUtils
{
    private const val USER_INFO_KEY = "userInfo"

    fun getUserInfo(): UserEntity?
    {
        val o = StpUtil.getSession().get(USER_INFO_KEY)
        if (o != null)
        {
            return o as UserEntity
        }
        return null
    }

    fun setUserInfo(userEntity: UserEntity)
    {
        StpUtil.getSession().set(USER_INFO_KEY, userEntity)
    }
}
